docs/index.md

```diff
@@ -459,6 +459,531 @@ class Settings(BaseSettings):
+`_cli_parse_args`. Alternatively, we could have set `_cli_parse_args=True` to parse args from the command line (i.e.,
+`sys.argv[1:]`).
+
+Lastly, a CLI settings source is always [**the topmost source**](#field-value-priority), and does not support [changing
+its priority](#changing-priority).
+
+#### Enable CLI Argument Parsing
+
+`cli_parse_args: Optional[Union[List[str], bool]] = None`
+
+* Default = `None`
+* If `True`, parse from `sys.argv[1:]`
+* If `List[str]`, parse from `List[str]`
+* If `False` or `None`, do not parse CLI arguments
+
+#### Lists
+
+CLI argument parsing of lists supports intermixing of any of the below three styles:
+
+  * JSON style `--field='[1,2]'`
+  * Argparse style `--field 1 --field 2`
+  * Lazy style `--field=1,2`
+
+```py
+from typing import List
+
+from pydantic_settings import BaseSettings
+
+
+class Settings(BaseSettings):
+    my_list: List[int]
+
+
+print(Settings(_cli_parse_args=['--my_list', '[1,2]']).model_dump())
+#> {'my_list': [1, 2]}
+
+print(Settings(_cli_parse_args=['--my_list', '1', '--my_list', '2']).model_dump())
+#> {'my_list': [1, 2]}
+
+print(Settings(_cli_parse_args=['--my_list', '1,2']).model_dump())
+#> {'my_list': [1, 2]}
+```
+
+#### Dictionaries
+
+CLI argument parsing of dictionaries supports intermixing of any of the below two styles:
+
+  * JSON style `--field='{"k1": 1, "k2": 2}'`
+  * Environment variable style `--field k1=1 --field k2=2`
+
+These can be used in conjunction with list forms as well, e.g:
+
+  * `--field k1=1,k2=2 --field k3=3 --field '{"k4: 4}'` etc.
+
+```py
+from typing import Dict
+
+from pydantic_settings import BaseSettings
+
+
+class Settings(BaseSettings):
+    my_dict: Dict[str, int]
+
+
+print(Settings(_cli_parse_args=['--my_dict', '{"k1":1,"k2":2}']).model_dump())
+#> {'my_dict': {'k1': 1, 'k2': 2}}
+
+print(Settings(_cli_parse_args=['--my_dict', 'k1=1', '--my_dict', 'k2=2']).model_dump())
+#> {'my_dict': {'k1': 1, 'k2': 2}}
+```
+
+### Subcommands and Positional Arguments
+
+Subcommands and positional arguments are expressed using the `CliSubCommand` and `CliPositionalArg` annotations. These
+annotations can only be applied to required fields (i.e. fields that do not have a default value). Furthermore,
+subcommands must be a valid type derived from the pydantic `BaseModel` class.
+
+!!! note
+    CLI settings subcommands are limited to a single subparser per model. In other words, all subcommands for a model
+    are grouped under a single subparser; it does not allow for multiple subparsers with each subparser having its own
+    set of subcommands. For more information on subparsers, see [argparse
+    subcommands](https://docs.python.org/3/library/argparse.html#sub-commands).
+
+```py test="skip"
+from pydantic import BaseModel
+
+from pydantic_settings import (
+    BaseSettings,
+    CliPositionalArg,
+    CliSubCommand,
+)
+
+
+class FooPlugin(BaseModel, use_attribute_docstrings=True):
```

Comment:
rather than using `use_attribute_docstrings=True` everywhere, we would be better of using `Field(description='...')` in all but one example.

Reply:
Sure, will update.

Reply:
I ended removing all instances of `use_attribute_docstrings`. The user should know from Pydantic if they want to enable this or not.